src/main/java/org/apache/sling/graphql/core/servlet/GraphQLServlet.java

```diff
@@ -196,8 +196,11 @@ private void activate(Config config) {
         cacheMisses = metricsService.counter(METRIC_NS + "." + servletRegistrationProperties + ".cache_misses");
         requestsServed = metricsService.counter(METRIC_NS + "." + servletRegistrationProperties + ".requests_total");
         gaugeCacheHitRate = METRIC_NS + "." + servletRegistrationProperties + ".cache_hit_rate";
-        metricRegistry.register(gaugeCacheHitRate,
-                (Gauge<Float>) () -> (cacheHits.getCount() / (float) (cacheHits.getCount() + cacheMisses.getCount())));
+        metricRegistry.register(gaugeCacheHitRate, (Gauge<Float>) () -> {
+            float hitCount = cacheHits.getCount();
+            float missCount = cacheMisses.getCount();
+            return hitCount > 0 || missCount > 0 ? hitCount / (hitCount + missCount) : 0.0f;
```

Comment:
I think testing for `missCount > 0` is sufficient to avoid NaN but that's a detail